File: Ejercicio4.py
#Ejercicio 4: Arreglos paralelos de 25 sucursales y sus ventas.
import time
import random

print("--- Ejercicio 4: Sucursales y Ventas ---")

# Inicia cronómetro (incluye la generación de los arreglos y los cálculos)
start_time = time.time()

# Generar datos simulados
sucursales = [f"Sucursal {i+1}" for i in range(25)]
ventas = [random.uniform(1000, 5000) for _ in range(25)]

# Cálculos
suma_ventas = sum(ventas)
promedio_ventas = suma_ventas / len(ventas)

print(f"El promedio general de ventas es: ${promedio_ventas:.2f}")
print("Sucursales por encima del promedio:")

for i in range(25):
    if ventas[i] > promedio_ventas:
        print(f"- {sucursales[i]}: ${ventas[i]:.2f}")

# Detiene cronómetro
end_time = time.time()
print(f"Tiempo de ejecución: {end_time - start_time:.6f} segundos\n")

File: Clase 09-03-2026 - Actividad 1/Ejercicio4.cpp
//Ejercicio 4: Arreglos paralelos de 25 sucursales y sus ventas.

#include <iostream>
#include <chrono>
#include <cstdlib>
#include <ctime>
#include <string>

using namespace std;
using namespace std::chrono;

int main() {
    cout << "--- Ejercicio 4: Sucursales y Ventas ---\n";
    srand(time(NULL));

    // Inicia cronómetro
    auto start = high_resolution_clock::now();

    string sucursales[25];
    float ventas[25];
    float suma_ventas = 0;

    // Generar datos simulados y sumar
    for(int i = 0; i < 25; i++) {
        sucursales[i] = "Sucursal " + to_string(i + 1);
        ventas[i] = (rand() % 4001) + 1000;
        suma_ventas += ventas[i];
    }

    float promedio_ventas = suma_ventas / 25.0;

    cout << "El promedio general de ventas es: $" << promedio_ventas << "\n";
    cout << "Sucursales por encima del promedio:\n";

    for(int i = 0; i < 25; i++) {
        if(ventas[i] > promedio_ventas) {
            cout << "- " << sucursales[i] << ": $" << ventas[i] << "\n";
        }
    }

    // Detiene cronómetro
    auto end = high_resolution_clock::now();
    duration<double> time_span = duration_cast<duration<double>>(end - start);
    
    cout << "Tiempo de ejecucion: " << time_span.count() << " segundos\n\n";
    return 0;
}